libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -1151,10 +1154,13 @@ void AP_DDS_Client::update()
     }
 
     if (cur_time_ms - last_battery_state_time_ms > DELAY_BATTERY_STATE_TOPIC_MS) {
-        constexpr uint8_t battery_instance = 0;
-        update_topic(battery_state_topic, battery_instance);
+        for(uint8_t battery_instance = 0; battery_instance < AP_BATT_MONITOR_MAX_INSTANCES; battery_instance++){
+            update_topic(battery_state_topic, battery_instance);
+            if (battery_state_topic.present){
```

Comment:
What happened to checking the type is enabled?

Reply:
it's not necessary, only the healthy batteries are published, thanks to `battery.healthy(instance)`

Reply:
```
ros2 topic echo /ap/battery --field header
stamp:
  sec: 1727912371
  nanosec: 797837000
frame_id: '0'
---
stamp:
  sec: 1727912372
  nanosec: 800769000
frame_id: '0'
---
stamp:
  sec: 1727912373
  nanosec: 803701000
frame_id: '0'
---
stamp:
  sec: 1727912374
  nanosec: 813297000
frame_id: '0'
---
stamp:
  sec: 1727912375
  nanosec: 814563000
frame_id: '0'
---
stamp:
  sec: 1727912376
  nanosec: 819994000
frame_id: '0'
---
stamp:
  sec: 1727912377
  nanosec: 822926000
frame_id: '0'
---
```

Reply:
I pushed a new test that verifies that only battery '0' is published, which is the only available for the SITL.